src/conn/conn_api.c

```diff
@@ -1643,6 +1643,45 @@ __wt_verbose_config(WT_SESSION_IMPL *session, const char *cfg[])
 }
 
 /*
+ * __conn_confchk --
+ *	Check for valid configure option combinations. Individual options
+ *	are checked as they are parsed. This function is a place to check
+ *	combinations of options across subsystems.
+ */
+static int
+__conn_confchk(WT_SESSION_IMPL *session)
+{
+	WT_CONNECTION_IMPL *conn;
+
+#define	WT_CONN_CONFCHK_PREFIX	"In memory configuration incompatible with "
+
+	conn = S2C(session);
+
+	/*
+	 * In memory configuration requires specific settings for a number
+	 * of other configuration values. Check those now.
+	 */
+	if (F_ISSET(conn, WT_CONN_IN_MEMORY)) {
+		if (FLD_ISSET(conn->log_flags, WT_CONN_LOG_ENABLED))
+			WT_RET_MSG(session, EINVAL,
+			    WT_CONN_CONFCHK_PREFIX "log=(enabled=true)");
+		else if (conn->sweep_idle_time != 0)
+			WT_RET_MSG(session, EINVAL,
+			    WT_CONN_CONFCHK_PREFIX
+			    "non zero file_manager=(close_idle_time)");
+		else if (conn->ckpt_usecs != 0)
+			/*
+			 * It is not necessary to check for log based
+			 * automatic checkpoints since logging must be
+			 * disabled.
+			 */
+			WT_RET_MSG(session, EINVAL,
+			    WT_CONN_CONFCHK_PREFIX "checkpoints");
+	}
+	return (0);
+}
```

Comment:
@agorrod, I think each subsystem should be responsible for doing its own configuration validation, that is, `__wt_sweep_config()` should own checking if something else is configured that is a problem for the sweep configuration.

And, I think it should be a configuration check, that is, we should check for problematic configuration strings instead of checking some other subsystem's global configuration information. Arguably, `WT_CONN_IN_MEMORY` is global information, though, I don't mind checking it in other places.